### DISCUSSION TITLE
Idea: Scanner Redesign For Speed-Up

### DISCUSSION THREAD

**Discussion by loveemu** [vgmtrans/vgmtrans#273] [General]
At present, a scanner parses the whole file, and the next one will do the same, and so on.

A scanner should skip the region that is already found by other scanner. Also, all scanners should run parallelly. And it will give a chance to skip unneeded scanning to every scanners.

**Comment by loveemu** [comment]
### Idea: Add Classifier Class To Help Scanning

Who wants to search PSX ADPCMs from GBA ROM?
Probably, we are currently doing several unwanted scans.

How about testing the console type by a simple class before scanning?
For example, a scanner can skip its test routine if the input is classified to FILE_TYPE_GBA.

It might not make scanning much faster, if scanners can quit their test quickly enough.
However, I think it is somewhat useful for scanners that often raises false-positive. (PSXSampColl, for instance)

**Comment by loveemu** [comment]
I will move the issue to discussion as I think this issue is out of the realm of epic level ideas.

**Reply by Meerkov** [reply]
Doesn't the classifier already claim different sectors?
